apps/worker/src/graph_publisher/graph.publisher.processor.service.ts

```diff
@@ -1,31 +1,119 @@
+import { BlockchainService } from '../../../../libs/common/src/blockchain/blockchain.service';
+import { createKeys } from '../../../../libs/common/src/blockchain/create-keys';
+import { ITxMonitorJob } from '../../../../libs/common/src/dtos/graph.notifier.job';
+
+export const SECONDS_PER_BLOCK = 12;
 
 @Injectable()
 @Processor(QueueConstants.GRAPH_CHANGE_PUBLISH_QUEUE)
 export class GraphUpdatePublisherService extends BaseConsumer {
   constructor(
     @InjectRedis() private cacheManager: Redis,
+    @InjectQueue(QueueConstants.GRAPH_CHANGE_NOTIFY_QUEUE) private graphChangeNotifyQueue: Queue,
     private configService: ConfigService,
+    private blockchainService: BlockchainService,
+    private nonceService: NonceService,
   ) {
     super();
   }
 
+  /**
+   * Processes a job for graph update.
+   * @param job - The job to process.
+   * @returns A promise that resolves when the job is processed.
+   */
   async process(job: Job<GraphUpdateJob, any, string>): Promise<any> {
     this.logger.log(`Processing job ${job.id} of type ${job.name}`);
+    let statefulStorageTxHash: Hash = {} as Hash;
+    const lastFinalizedBlockHash = await this.blockchainService.getLatestFinalizedBlockHash();
+    const currentCapacityEpoch = await this.blockchainService.getCurrentCapacityEpoch();
     try {
-      // TODO: add logic to send update to Frequency stateful storage
-      this.logger.debug(job.asJSON());
+      switch (job.data.update.type) {
+        case 'PersistPage': {
+          let payloadData: number[] = [];
+          if (typeof job.data.update.payload === 'object' && 'data' in job.data.update.payload) {
+            payloadData = Array.from((job.data.update.payload as { data: Uint8Array }).data);
+          }
+          const providerKeys = createKeys(this.configService.getProviderAccountSeedPhrase());
+          const tx = this.blockchainService.createExtrinsicCall(
+            { pallet: 'statefulStorage', extrinsic: 'upsertPage' },
+            job.data.update.ownerDsnpUserId,
+            job.data.update.schemaId,
+            job.data.update.pageId,
+            job.data.update.prevHash,
+            payloadData,
+          );
+          statefulStorageTxHash = await this.processSingleBatch(providerKeys, tx);
+          break;
+        }
+        default:
+          break;
+      }
+
+      this.logger.debug(`job: ${JSON.stringify(job, null, 2)}`);
+
+      // Add a job to the graph change notify queue
+      const txMonitorJob: ITxMonitorJob = {
+        id: job.data.referenceId,
+        txHash: statefulStorageTxHash,
+        epoch: currentCapacityEpoch.toString(),
+        lastFinalizedBlockHash,
+        referencePublishJob: job.data,
+      };
+      const blockDelay = SECONDS_PER_BLOCK * MILLISECONDS_PER_SECOND;
+
+      this.graphChangeNotifyQueue.add(`Graph Change Notify Job - ${txMonitorJob.id}`, txMonitorJob, {
+        removeOnFail: false,
+        removeOnComplete: 2000,
+        delay: blockDelay,
+      });
```

Comment:
Nice, the happy path is working, we can try handling some errors like low capacity, and we can use `EventEmitter2` to send out a `capacity.low` event in case we encounter, in that case we can pause all the queues till next epoch (this can be another story later on)